script/deploy/facets/UpdateWhitelistManagerFacet.s.sol

```diff
@@ -269,212 +299,182 @@ contract DeployScript is UpdateScriptBase {
     }
 
     /// @notice Parses the ".DEXS" section of the whitelist JSON.
-    /// @dev Uses a two-pass strategy: first count, then allocate, then populate.
-    /// @param whitelistJson The raw JSON string.
-    /// @return contracts Array of addresses from the DEXS section.
-    /// @return selectors Array of selector arrays for each address.
     function _parseDexsSection(
         string memory whitelistJson
     )
         internal
+        view
         returns (address[] memory contracts, bytes4[][] memory selectors)
     {
-        // 1. Count total contracts for this network to pre-allocate arrays
-        uint256 totalContracts = _countDexsContracts(whitelistJson);
+        // Parse JSON once using JSONParserLib
+        JSONParserLib.Item memory root = JSONParserLib.parse(whitelistJson);
+        // prettier-ignore
+        JSONParserLib.Item memory dexs = JSONParserLib.at(root, "\"DEXS\"");
+
+        (uint256 totalContracts, uint256 totalDexs) = _countDexsContracts(
+            dexs
+        );
 
-        // 2. Allocate arrays
         contracts = new address[](totalContracts);
         selectors = new bytes4[][](totalContracts);
 
-        // 3. Populate arrays with data
         uint256 contractIndex = 0;
         contractIndex = _populateDexsContracts(
-            whitelistJson,
+            dexs,
             contracts,
             selectors,
-            contractIndex
+            contractIndex,
+            totalDexs
         );
     }
 
     /// @notice Parses the ".PERIPHERY" section of the whitelist JSON.
-    /// @dev Uses a two-pass strategy: first count, then allocate, then populate.
-    /// @param whitelistJson The raw JSON string.
-    /// @return contracts Array of addresses from the PERIPHERY section.
-    /// @return selectors Array of selector arrays for each address.
     function _parsePeripherySection(
         string memory whitelistJson
     )
         internal
+        view
         returns (address[] memory contracts, bytes4[][] memory selectors)
     {
-        // Check if PERIPHERY section exists for current network
-        string memory peripheryKey = string.concat(".PERIPHERY.", network);
-        try vm.parseJson(whitelistJson, peripheryKey) returns (bytes memory) {
-            // 1. Count contracts
-            uint256 totalContracts = _countPeripheryContracts(whitelistJson);
-
-            // 2. Allocate arrays
-            contracts = new address[](totalContracts);
-            selectors = new bytes4[][](totalContracts);
-
-            // 3. Populate arrays
-            uint256 contractIndex = 0;
-            contractIndex = _populatePeripheryContracts(
-                whitelistJson,
-                contracts,
-                selectors,
-                contractIndex
-            );
-        } catch {
-            // Network not found, throw error
+        // Parse JSON once using JSONParserLib
+        JSONParserLib.Item memory root = JSONParserLib.parse(whitelistJson);
```

Comment:
we are doing this operation two times (once for periphery and once for DEXs. 
Wondering about the runtime/load. Maybe makes sense to push this to a storage variable and reuse?
But I may be wrong there...